traffic_portal/app/src/common/modules/table/deliveryServiceJobs/TableDeliveryServiceJobsController.js

```diff
@@ -35,7 +35,7 @@ var TableDeliveryServiceJobsController = function(deliveryService, jobs, $contro
 			return $scope.deliveryService.xmlId;
 		},
 		getHref: function () {
-			return "#!/delivery-services/" + $scope.deliveryService.id + "?type=" + $scope.deliveryService.type;
+			return encodeURI("#!/delivery-services/" + $scope.deliveryService.id + "?type=" + $scope.deliveryService.type);
```

Comment:
That won't work, `encodeURI` expects a full URI as input, and because we are putting query string parameters in x-www-form-urlencoded format inside a document fragment, it assumes nothing needs to be escaped and leaves the input unchanged:
![image](https://user-images.githubusercontent.com/6013378/126548404-c12c5e6e-c4bf-4f64-8992-ce3d4ed02e8a.png)

Using `encodeURIComponent` expects the entirety of its input to be a single component of a URI, so it will encode too much:

![image](https://user-images.githubusercontent.com/6013378/126548478-7c3de528-1ae4-4152-a94b-719b2e260411.png)


You need to use `encodeURIComponent` only on the part that needs to be encoded as a single URI component, like:

```javascript
"#!/delivery-services/" + $scope.deliveryService.id + "?type=" + encodeURIComponent($scope.deliveryService.type);
```

Reply:
Already changed